sklearn/utils/validation.py

```diff
@@ -2148,9 +2155,10 @@ def _check_sample_weight(
         dtype of the validated `sample_weight`.
         If None, and `sample_weight` is an array:
 
-            - If `sample_weight.dtype` is one of `{np.float64, np.float32}`,
+            - If `sample_weight.dtype` is one of `{xp.float64, xp.float32}`,
```

Comment:
If we are not changing types in public functions, I wonder if we should keep private ones as is too, for consistency?